experimental/generation/generator/packages/library/src/dialogGenerator.ts

```diff
@@ -713,3 +736,34 @@ export async function generate(
         feedback(FeedbackType.error, e.message)
     }
 }
+
+/**
+ * Expand a property definition by resolving $ref including template:, removing allOf and generating $entities if missing.
+ * @param property Name of property.
+ * @param schema Schema definition.
+ * @param templateDirs Optional directories for overriding templates.
+ */
+export async function expandPropertyDefinition(property: string, schema: any, templateDirs?: string[]): Promise<any> {
```

Comment:
This could be a Schema type and the turn could be a Schema type. If the actual parameter has additional properties, the method should ensure they are not cleaved off on expansion.

Reply:
The schema type is the whole definition of JSON schema.  Or are you suggesting a named type of "any"?